### PR TITLE
CTCTRADER-2512 Update validation on Principal eori 

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/controllers/traderDetails/WhatIsPrincipalEoriController.scala [hunk 1]
```diff
@@ -48,13 +50,15 @@ class WhatIsPrincipalEoriController @Inject() (
...
     implicit request =>
+      val isSimplified = request.userAnswers.get(ProcedureTypePage).contains(Simplified)
+      val countryCode = request.userAnswers.get(OfficeOfDeparturePage) map (
+        x => x.countryId
+      )
       val preparedForm = request.userAnswers.get(WhatIsPrincipalEoriPage) match {
-        case None        => form
-        case Some(value) => form.fill(value)
+        case None        => formProvider(isSimplified, countryCode.getOrElse(CountryCode("")))
+        case Some(value) => formProvider(isSimplified, countryCode.getOrElse(CountryCode(""))).fill(value)
```
Comment (MatthewAlexanderBeck) [inline app/controllers/traderDetails/WhatIsPrincipalEoriController.scala:61]: Is there a dependency here for the OfficeOfDeparture page, would be rather fail than pass a empty countryCode if its empty?

File: app/controllers/traderDetails/WhatIsPrincipalEoriController.scala [hunk 2]
```diff
@@ -68,7 +72,14 @@ class WhatIsPrincipalEoriController @Inject() (
...
     implicit request =>
-      form
+      val isSimplified = request.userAnswers.get(ProcedureTypePage) match {
+        case Some(Simplified) => true
+        case _                => false
+      }
+      val countryCode = request.userAnswers.get(OfficeOfDeparturePage) map (
+        x => x.countryId
+      )
+      formProvider(isSimplified, countryCode.getOrElse(CountryCode("")))
```
Comment (MatthewAlexanderBeck) [inline app/controllers/traderDetails/WhatIsPrincipalEoriController.scala:82]: Same here

File: app/forms/mappings/Constraints.scala [hunk 3]
```diff
@@ -174,6 +174,23 @@ trait Constraints {
...
+    }
+
+  def prefix(countryCode: CountryCode, errorKey: String): Constraint[String] =
+    Constraint {
+      case str if str.toUpperCase.take(2) == countryCode.code.toUpperCase.take(2) =>
+        Valid
+      case _ =>
+        Invalid(errorKey, countryCode.code)
+    }
+
```
Comment (MatthewAlexanderBeck) [inline app/forms/mappings/Constraints.scala:193]: Super minor comment but you could probably do this using the regxp method e.g. ^XI[0-9a-zA-Z]{1,15} would do the same thing as above. Both ways work however.

Comment (MatthewAlexanderBeck) [reply app/forms/mappings/Constraints.scala:193]: Also is there any tests for these methods? Not sure if there is.

**Review by connor-irvine — APPROVED** [review]

**Review by MatthewAlexanderBeck — APPROVED** [review]

**Pull request merged**